src/main/webapp/authorization.js

```diff
@@ -1,56 +1,143 @@
-// GCP Generated Github OAuth configuration
-const config = {
+// GCP Generated GitHub OAuth configuration
+const firebaseConfig = {
   apiKey: 'AIzaSyAR88Giah8cCEAvT_zDSIREWvgIIAeS8yY',
   authDomain: 'step2020-279820.firebaseapp.com',
 };
 
 /**
- * Initialize firebase if firebase is not already initialized. This method can
- * be called as many times as we need without any side effects.
+ * The enum returned after authorization is toggled
+ * @typedef AuthorizationEnum 
+ * @property {boolean} isValid false if an error occurs during authorization
+ * true otherwise.
+ * @property {String} gitHubToken a GitHub API access token if user was 
+ * authorized null otherwise
  */
-function initializeFirebase() {
-  const firebase = getFirebase();
-  if (firebase.apps.length === 0) {
-    firebase.initializeApp(config);
-  }
-}
 
 /**
- * Get firebase
- * @return {Firebase}
+ * Object used for GitHub authorization and as a wrapper for Firebase.
+ * It handles Firebase initialization on instantiation and holds nifty
+ * functions for handling GitHub authorization. We are using a class to have
+ * access tokens persist for ease of use on the backend,
+ *
+ * Make sure to include the scripts below in any HTML file that uses this class.
+ *
+ * <script src="https://www.gstatic.com/firebasejs/7.15.5/firebase.js"></script>
+ * <script type="module" src="authorization.js"></script>
+ *
+ * NOTE: The authorization.js file location might change.
  */
-function getFirebase() {
-  return firebase; //eslint-disable-line
-}
+class GitHubAuthorizer {
 
+  /**
+   * Create a GitHubAuthorizer.
+   */
+  constructor() {
 
-/**
- * Function called when clicking the Login/Logout button.
- */
-function toggleSignIn() {
-  const firebase = getFirebase();
-  if (!firebase.auth().currentUser) {
-    let provider = new firebase.auth.GithubAuthProvider();
-    firebase.auth().signInWithPopup(provider).then(function(result) {
-      // This gives you a GitHub Access Token.
-      // You can use it to access the GitHub API.
-      const token = result.credential.accessToken; //eslint-disable-line
-      // The signed-in user info.
-      const user = result.user; //eslint-disable-line
-    }).catch(function(error) {
-      // Handle Errors here.
-      console.error(error);
-      // Change this in future iterations
-      alert(error.message);
-    });
-  } else {
-    // if logged in, sign out
-    firebase.auth().signOut();
+    /** @type {string} GitHub API token */
+    this.token = null;
+
+    /**
+     * the Firebase instance associated with a given
+     * GitHubAuthorizer
+     * @type {Firebase}
+     */
+    this.firebase = firebase; // eslint-disable-line
+
+    this.initializeFirebase();
```

Comment:
Also, there is a bug with this class where if the user is signed in from another session, they identify as logged in, but because of how the auth token is assigned in this class the auth token is still null. I propose that we sign out any previous authentication sessions in the constructor. This works with the sign up flow I'm currently working on:
```suggestion
   /**
     * the firbase instance associated with a given 
     * GitHubAuthorizer
     * @type {Firebase} 
     */
    this.firebase = firebase; // eslint-disable-line

    this.initializeFirebase()

    /** @type {string} gitHub API token */
    this.token = null;

    // Ensure that user is not signed in from another session.
    this.signOut();
```